File: 백준/Silver/30804. 과일 탕후루/과일 탕후루.py
import sys
input = sys.stdin.readline

n = int(input())
arr = list(map(int, input().split()))

l, r, cnt, kind, ans = 0, 0, 0, 0, 0
nums = [0] * 10

while r < n:
    if nums[arr[r]] == 0:
        kind += 1
    cnt += 1
    nums[arr[r]] += 1
    
    if kind > 2:
        nums[arr[l]] -= 1
        if nums[arr[l]] == 0:
            kind -= 1
        l += 1
        cnt -= 1
    
    ans = max(cnt, ans)
    r += 1

print(ans)

File: 백준/Silver/30804. 과일 탕후루/과일 탕후루.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();

        int l = 0, r = 0, kind = 0, cnt = 0, ans = 0;
        int[] nums = new int[10];

        while (r < n) {
            if (++nums[arr[r]] == 1)
                kind++;
            cnt++;

            if (2 < kind) {
                if (--nums[arr[l]] == 0)
                    kind--;
                cnt --;
                l ++;
            }

            ans = Math.max(ans, cnt);
            r ++;
        }

        System.out.println(ans);
    }
}